coderd/database/dbtestutil/postgres.go

```diff
@@ -59,28 +66,38 @@ func initDefaultConnection(t TBSubset) error {
 		DBName:   "postgres",
 	}
 	dsn := params.DSN()
-	db, dbErr := sql.Open("postgres", dsn)
-	if dbErr == nil {
-		dbErr = db.Ping()
-		if closeErr := db.Close(); closeErr != nil {
-			return xerrors.Errorf("close db: %w", closeErr)
+
+	// Helper closure to try opening and pinging the default Postgres instance.
+	// Used within a single retry loop that handles both retryable and permanent errors.
+	attemptConn := func() error {
+		db, err := sql.Open("postgres", dsn)
+		if err == nil {
+			err = db.Ping()
+			if closeErr := db.Close(); closeErr != nil {
+				return xerrors.Errorf("close db: %w", closeErr)
+			}
 		}
+		return err
 	}
-	shouldOpenContainer := false
-	if dbErr != nil {
-		errSubstrings := []string{
-			"connection refused",          // this happens on Linux when there's nothing listening on the port
-			"No connection could be made", // like above but Windows
+
+	var dbErr error
+	// Retry up to 3 seconds for temporary errors.
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
```

Comment:
It may be a good idea to raise the timeout a little. I worry that it might take a bit longer for postgres to start accepting connections.

```suggestion
	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```